### DISCUSSION TITLE
Pre-runtime validation

### DISCUSSION THREAD

**Discussion by JSteinbauer** [dagster-io/dagster#10334] [General]
It would be nice to have "pre-runtime validation" - is this already supported?

Example.:
Op "load_trips" has the DagsterType output "TripsDataFrame", but the downstream Op "generate_plot" takes the DagsterType input "TripsDataFrame_2". 

Obviously, this will cause a runtime error, but why would we have to wait for Op "load_trips" to finish executing before the program crashes?!

Best,
Jakob

**Comment by sryza** [comment]
Hey @JSteinbauer - Dagster doesn't currently support this, but we're tracking it here: https://github.com/dagster-io/dagster/issues/9847.
